spec/scenic/schema_dumper_spec.rb

```diff
@@ -36,4 +36,48 @@ class Search < ActiveRecord::Base; end
       Search.connection.drop_view :'scenic.searches'
     end
   end
+
+  context "with dependent views " do
+    it "dumps create_view sorted alphabetically" do
+      Search.connection.execute("CREATE SCHEMA scenic")
+      Search.connection.execute("SET search_path TO scenic, public")
+
+      views = {
+        "scenic.view1": "SELECT 'needle1'::text AS haystack",
+        "scenic.view2": "SELECT 'needle2'::text AS haystack",
+        "scenic.view1_1": "SELECT * FROM scenic.view1",
+        "scenic.view1_2": "SELECT * FROM scenic.view1",
+      }
+
+      expected_order = %w(
+        scenic.view1
+        scenic.view1_1
+        scenic.view1_2
+        scenic.view2
+      )
+
+      views.each do |name, definition|
+        Search.connection.create_view(name, sql_definition: definition)
+      end
+
+      stream = StringIO.new
+
+      ActiveRecord::SchemaDumper.dump(Search.connection, stream)
+
+      output = stream.string
+
+      views.each do |name, _|
+        expect(output).to include "create_view :'#{name}',"
+      end
+
+      create_views = output.lines.grep(/create_view :/)
+      order = create_views.map { |line| line.scan(/scenic\.[a-z_0-9]+/).first }
```

Comment:
`\w` covers the group you want: `/\w/ - A word character ([a-zA-Z0-9_])` So I think this can be `/scenic\.\w+/`.